tsgen/__main__.py

```diff
@@ -100,13 +100,15 @@ def generate_on_car_telemetry(self):
                 #define CAN_HANDLERS_H
 
                 #include <stdint.h>
+                #include <stddef.h>
 
                 /**
                  * @brief   Entry in CAN handler table
                  */
                 typedef struct {
                     uint32_t identifier;
                     void (*unpack_func)(uint8_t*, const uint8_t*, size_t);
+                    void (*pack_func)(uint8_t*, const uint8_t*, size_t);
```

Comment:
I'm hesitant to add every single packing function to the table because the compiler will not be able to remove ANY of these functions from the text of the binary, even if they are unused. Since these functions aren't used by the actual on-car telemetry implementation and it's a lot of code, it's quite wasteful of program memory.

I'm not even sure that adding the packing functions to the look-up table has any real benefit, so let me go through it and we can discuss.

The process a received message goes through is:
- CAN message arrives.
- A search of the handler table is made to identify which unpacking function should be called for that message ID.
- The wrapper function is called, which converts a pointer to the payload area of the telemetry PDU into a pointer to whatever `struct` type the unpacking function has.
- The unpacking function is called which will re-structure the data in the CAN message to the format described by its corresponding `struct`, and write that structure into the telemetry PDU.
- The PDU is then transmitted wirelessly and received by the intermediate server.
- The intermediate server uses the definitions in its schema to extract the sensor data from the structs.

Now think about going the other way where you have raw sensor data and need to pack it into a CAN message:
- You start with raw sensor data, either generated by a function or read out from a look-up table.
- In order to call the packing function, you need to put your sensor data into the `struct` format for a particular message.
- You need to know what the type of that struct is in order to add data to it, and the code to add data to the struct cannot be generalised (at least not easily). Therefore, you need to manually write code to put all the raw data into the struct.
- At that point, you know what the `struct` is therefore you can just call the packing function. There's no real benefit in automating that process, especially given that it adds so much code size to the actual on-car telemetry implementation.

I think there is a much nicer way to do this, which is to use Python and `cantools` to generate pre-packed test vectors for each CAN message rather than doing the emulation and packing on the embedded device itself. All you would need to do is store a big table for each CAN message you want to include in the emulation and just read through that table on the embedded side. You could generate the test CAN messages using the emulators in the intermediate server schema, or use real logged data once we have it. This approach also means that the microcontroller has to do far less work, which should help reduce the overhead of the testbench.